### PR TITLE
Feature: https only -- adds option to enforce https on running instances

### PR DESCRIPTION
**Pull request by benbusby**
Enforces https by default on Docker instances, adds an option to enforce https if running through pip/pipx.

### CODE REVIEW

**Comment by benbusby** [comment]
@FoxxMD @chpatton013 (or anyone else who comes across this and is a regular Docker user), do you have any input on this new feature? It basically just forces the Flask server to reroute to HTTPS by default if running via Docker, but instances run through other methods require a flag to enforce this behavior.

I'm a bit torn since I don't really like having different deployment options behave differently by default, but it solves a problem brought up in #17 where the Flask server (believing that it's running as HTTP only) would reroute to the non-HTTPS variety of whatever URL is being used. If the site wasn't configured properly, it either would get stuck on the HTTP connection from this point on, or just hang if port 80 wasn't opened. 

Overall I think HTTPS enforcement is a good addition to solve issues like this, but I might be overlooking some cases where this would be a breaking change. Let me know your thoughts (if you have any, feel free to ignore if you're indifferent about it).

**Comment by FoxxMD** [comment]
First off, it's configurable so that's good 👍

WRT deployment options I agree with you that they should be consistent. If the default using pip is not to enforce then I don't think you should enforce in docker either. IMO this is a configuration setting that should be left up to the user to use correctly. If you feel like users need guidance on using it correctly you could always add some info to the readme, however.

As far as breaking changes this (enforcing https by default) would definitely break deployments for anyone who is already running the docker and using it behind a reverse proxy (like me).

**Comment by FoxxMD** [comment]
Issue adjacent -- allowing docker users to specify the port whoogle runs on via ENV variable would be much appreciated. An [unraid user](https://forums.unraid.net/topic/92360-support-foxxmd-whoogle-search/?do=findComment&comment=855506) asked me about running on port 443 in a configuration (in unraid) that doesn't allow us to map ports between container/host so they have no real recourse ATM. Would be an easy fix I think ;)

**Comment by chpatton013** [comment]
Agreed with @FoxxMD; Keep it consistent and configurable. I also run all my docker apps behind a reverse-proxy, so I prefer them to do dumb HTTP by default.

Adnn @FoxxMD WRT your request for a configurable port variable, I was actually just about to open a PR to do that.

**Comment by benbusby** [comment]
Great points, I hadn't thought of the breaking changes to reverse proxies yet, so I'm glad I got a second (and now third) set of eyes on this. 

I just updated this to default to HTTP, and snuck in an update to set a configurable port variable as well since they're related.

File: Dockerfile [hunk 1]
```diff
@@ -10,8 +10,14 @@ RUN mkdir $config_dir
 VOLUME $config_dir
 ENV CONFIG_VOLUME=$config_dir
 
+ARG use_https=''
+ENV HTTPS_ONLY=$use_https
+
+ARG whoogle_port=5000
+ENV EXPOSE_PORT=${EXPOSE_PORT:-$whoogle_port}
```
Comment (chpatton013) [inline Dockerfile:17]: AFAIK the use of `EXPOSE_PORT` on the RHS of this assignment isn't helping you. You effectively have an empty env at image build time, and you haven't set any ARGs or ENVs for `EXPOSE_PORT` before this line. I think it's equivalent just to do `ENV EXPOSE_PORT=$whoogle_port`

Comment (benbusby) [reply Dockerfile:17]: Ah crap, yeah I was messing around with a few things and didn't mean to keep that change. Thanks for pointing that out, will remove.

**Comment by chpatton013** [comment]
:shipit:

**Pull request merged**